chromium/browser_type.go

```diff
@@ -134,26 +136,46 @@ func (b *BrowserType) Launch(opts goja.Value) api.Browser {
 		envs = append(envs, fmt.Sprintf("%s=%s", k, v))
 	}
 
+	logger, err := makeLogger(b.Ctx, launchOpts)
+	if err != nil {
+		k6common.Throw(rt, fmt.Errorf("cannot make logger: %w", err))
+	}
+
+	flags := prepareFlags(launchOpts, &state.Options)
+
+	finalDirPath, cleanupFunc, err := b.createTempDir(flags["user-data-dir"], logger)
+	if err != nil {
+		k6common.Throw(rt, err)
+	}
+	flags["user-data-dir"] = finalDirPath
+
+	go func(ctx context.Context) {
+		defer cleanupFunc()
+		// There's a small chance that this might be called
+		// if the context is closed by the k6 runtime. To
+		// guarantee the cleanup we would need to orchestrate
+		// it correctly which https://github.com/grafana/k6/issues/2432
+		// will enable once it's complete.
+		<-ctx.Done()
```

Comment:
Good point. From my tests it seemed to be called pretty consistently, as there's apparently enough time between [when the context is cancelled](https://github.com/grafana/k6/blob/b725f03be5873b9b0d933b94a83924ff7c1f15e4/js/runner.go#L746-L748) and k6 shutdown for `cleanupFunc()` to finish running. And it worked fine with multiple iterations and VUs. But it's probably a "works fine on my machine" scenario :sweat_smile:

Reply:
Created an issue to track the fix to make deletion deterministic i.e. use the fix from grafana/k6#2432: grafana/xk6-browser#403